pinc/showavailablebooks.inc

```diff
@@ -384,14 +390,15 @@ function show_project_listing(
         SELECT *,
             (unix_timestamp()    - modifieddate    )/(24 * 60 * 60) AS days_avail,
             (smoothread_deadline - unix_timestamp())/(24 * 60 * 60) AS days_left
-        FROM projects
+        FROM projects NATURAL JOIN genres
         $optional_join_clause
         WHERE
             $filtered_project_selector
         ORDER BY
             $presort
             $curr_sql_orderclause
     ";
+    mysqli_report(MYSQLI_REPORT_ERROR | MYSQLI_REPORT_STRICT);
```

Comment:
We should think if there is a compelling reason to not just make this a global setting in `base.inc`.

Reply:
The manual says PHP versions since 5.3.4 and 5.2.15 "Changing the reporting mode is now be per-request, rather than per-process." so we mght not be able to do that.

Reply:
From a per-request vs per-process perspective either we can do this here and in base.inc or we can't do it at all. The question is really do we want all of our `mysqli()` calls to have this behavior or just the ones that happen after this line in the code?

Reply:
I think I completely misunderstood what "per-request, rather than per-process" means. I still don't know what it means but if putting the mysqli_report line in base.inc would affect all subsequent mysqli_querys I'm all for it.

Reply:
I've put it in base.inc. It works.

Reply:
Right, I know it will work there for what you are doing, the question is: will this inadvertently mess up anything else. In other words: are there places in the code that we want to do something different than raise/die on a failed mysqli call. I think the answer is probably yes and we need to wrap those in a try/except before we put this in base.inc.

Reply:
Agreed.